### DISCUSSION TITLE
Connecting rigid bodies with spherical joints

### DISCUSSION THREAD

**Discussion by shamilmamedov** [stack-of-tasks/pinocchio#1834] [Q&A] [answered]
Hi everyone. I am trying to model a serial chain robot consisting of rigid bodies connected via spherical joints. There is no way to select spherical joint in URDF even though Pinocchio supports spherical joints. Would it be possible to create URDF with revolute joints and then change it to spherical in Pinocchio after building the model using `buildModelFromUrdf(urdf)`? Is there  a better way to create such a robot?

**Comment by jcarpent** [comment, marked as answer]
I would recommend building this robot programmaticaly, following this example: 
https://github.com/stack-of-tasks/pinocchio/blob/9719bdc0d3dc330df1ef53a080602c675b352a05/examples/simulation-pendulum.py

**Reply by shamilmamedov** [reply]
Thank you for quick reply! Quick related question: can I implement universal joint using `JointModelComposite`?

**Reply by jcarpent** [reply]
Yes, indeed. We plan to add it to Pinocchio soon.
